apiserver/common/tools.go

```diff
@@ -299,8 +299,12 @@ func (f *ToolsFinder) matchingStorageTools(args params.FindToolsParams) (coretoo
 	}
 	list := make(coretools.List, len(allMetadata))
 	for i, m := range allMetadata {
+		vers, err := version.ParseBinary(m.Version)
+		if err != nil {
+			return nil, err
```

Comment:
I'd use Annotatef here, perhaps fmt.Annotatef(err, "unexpectedly bad version %q in tools storage", m.Version).